packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -208,6 +209,16 @@ class DataSourceWithBackend<
 
     let url = '/api/ds/query?ds_type=' + this.type;
 
+    // Use the new query service for explore
+    if (config.featureToggles.queryServiceFromExplore && request.app === CoreApp.Explore) {
+      // make sure query-service is enabled on the backend
+      if (!(config.featureToggles.queryService || config.featureToggles.grafanaAPIServerWithExperimentalAPIs)) {
+        console.warn('feature toggle queryServiceFromExplore also requires the queryService to be running');
+      } else {
```

Comment:
nit: the negation in the if statement is really convenient but I find it hard to read so could this become more readable
```ts
const isQueryServiceEnabled = config.featureToggles.queryService;
const isExperimentalAPIsEnabled = config.featureToggles.grafanaAPIServerWithExperimentalAPIs;

if (!isQueryServiceEnabled && !isExperimentalAPIsEnabled) {
```

Reply:
good idea, applied, thanks 👍